hazelcast/src/main/java/com/hazelcast/quorum/QuorumServiceImpl.java

```diff
@@ -0,0 +1,290 @@
+    private ConcurrentMap<String, QuorumResolver> quorumResolvers = new ConcurrentHashMap<String, QuorumResolver>();
+    private ConcurrentMap<String, QuorumCallback> quorumCallbacks = new ConcurrentHashMap<String, QuorumCallback>();
+    private ConstructorFunction<String, QuorumCallback> cf = new ConstructorFunction<String, QuorumCallback>() {
+        @Override
+        public QuorumCallback createNew(String quorumName) {
+            return new QuorumCallbackImpl(nodeEngine, quorumName);
+        }
+    };
+
+
+    public QuorumServiceImpl(NodeEngineImpl nodeEngine) {
+        this.nodeEngine = nodeEngine;
+        this.eventService = nodeEngine.getEventService();
+        Node node = nodeEngine.getNode();
+        this.quorumConfigs = node.getConfig().getQuorumConfigs();
+        initializeQuorums();
+        initializeListeners();
+    }
+
+    private void initializeQuorums() {
+        for (QuorumConfig quorumConfig : quorumConfigs.values()) {
+            initializeDefaultQuorumResolver(quorumConfig);
+        }
+    }
+
+
+    private void initializeDefaultQuorumResolver(QuorumConfig quorumConfig) {
+        String quorumName = quorumConfig.getName();
+        getOrPutSynchronized(quorumCallbacks, quorumName, quorumCallbacks, cf);
+        int size = quorumConfig.getSize();
+        if (size > 0) {
+            MemberCountQuorumResolver resolver = new MemberCountQuorumResolver(size);
+            quorumResolvers.put(quorumName, resolver);
+        }
+    }
+
+    private void initializeListeners() {
+        for (Map.Entry<String, QuorumConfig> configEntry : quorumConfigs.entrySet()) {
+            QuorumConfig config = configEntry.getValue();
+            for (QuorumListenerConfig listenerConfig : config.getListenerConfigs()) {
+                if (listenerConfig != null) {
+                    QuorumListener listener = null;
+                    if (listenerConfig.getImplementation() != null) {
+                        listener = listenerConfig.getImplementation();
+                    } else if (listenerConfig.getClassName() != null) {
+                        try {
+                            listener = ClassLoaderUtil
+                                    .newInstance(nodeEngine.getConfigClassLoader(), listenerConfig.getClassName());
+                        } catch (Exception e) {
+                            throw ExceptionUtil.rethrow(e);
+                        }
+                    }
+                    if (listener != null) {
+                        String instanceName = configEntry.getKey();
+                        addQuorumListener(instanceName, listener);
+                    }
+                }
+            }
+        }
+    }
+
+    public void addQuorumListener(String name, QuorumListener listener) {
+        eventService.registerLocalListener(SERVICE_NAME, name, listener);
+    }
+
+
+    public void resolveQuorum(Operation op) {
```

Comment:
I would call this method 'checkQuorumObtained' or 'verifyQuorumObtained' The fact that you need to resolve something is not relevant to the end user. 

I think it would even be better if you return a boolean and let the OperationRunnerImpl decide what to do. Then the method can be called 'isQuorumObtained'.